### ISSUE TITLE
Cannot get albumentations augmentations to work with mask_rcnn_50 on a custom dataset

### ISSUE THREAD

**Issue by AndreaPi** [open-mmlab/mmdetection#6126]
**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
I try to train a mask_rcnn_50 model on a custom dataset of `COCODataset` type using Albumentation augmentations, following the Albu example in MMDetection repository:

https://github.com/open-mmlab/mmdetection/blob/master/configs/albu_example/mask_rcnn_r50_fpn_albu_1x_coco.py

but I get a `ValueError`.

**Reproduction**

My config file, below, is very similar to the Albu example. The main differences are:

- `img_norm_cfg` is different because ofc the mean and std of my custom dataset are different
-   the `img_scale` parameter in `dict(type='Resize', img_scale=(1920, 1440), keep_ratio=True)` has a different value
- I added a `val_pipeline` and `test_pipeline` to the existing `train_pipeline`
- I have 5 classes, instead of 80 as in the standard Microsoft COCO dataset

```python
# base config
_base_ = '/mmdetection/configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.py'

# pipelines setting
img_norm_cfg = dict(
    mean=[107.79, 115.57, 117.07], std=[48.47, 54.44, 56.12], to_rgb=False)
albu_train_transforms = [
    dict(
        type='ShiftScaleRotate',
        shift_limit=0.0625,
        scale_limit=0.0,
        rotate_limit=0,
        interpolation=1,
        p=0.5),
    dict(
        type='RandomBrightnessContrast',
        brightness_limit=[0.1, 0.3],
        contrast_limit=[0.1, 0.3],
        p=0.2),
    dict(
        type='OneOf',
        transforms=[
            dict(
                type='RGBShift',
                r_shift_limit=10,
                g_shift_limit=10,
                b_shift_limit=10,
                p=1.0),
            dict(
                type='HueSaturationValue',
                hue_shift_limit=20,
                sat_shift_limit=30,
                val_shift_limit=20,
                p=1.0)
        ],
        p=0.1),
    dict(type='JpegCompression', quality_lower=85, quality_upper=95, p=0.2),
    dict(type='ChannelShuffle', p=0.1),
    dict(
        type='OneOf',
        transforms=[
            dict(type='Blur', blur_limit=3, p=1.0),
            dict(type='MedianBlur', blur_limit=3, p=1.0)
        ],
        p=0.1),
]
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
    dict(type='Resize', img_scale=(1920, 1440), keep_ratio=True),
    dict(type='Pad', size_divisor=32),
    dict(
        type='Albu',
        transforms=albu_train_transforms,
        bbox_params=dict(
            type='BboxParams',
            format='coco',
            label_fields=['gt_labels'],
            min_visibility=0.0,
            filter_lost_elements=True),
        keymap={
            'img': 'image',
            'gt_masks': 'masks',
            'gt_bboxes': 'bboxes'
        },
        update_pad_shape=False,
        skip_img_without_anno=True),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='DefaultFormatBundle'),
    dict(
        type='Collect',
        keys=['img', 'gt_bboxes', 'gt_labels', 'gt_masks'],
        meta_keys=('filename', 'ori_shape', 'img_shape', 'img_norm_cfg',
                   'pad_shape', 'scale_factor'))
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
    dict(type='Resize', img_scale=(1920, 1440), keep_ratio=True),
    dict(type='Pad', size_divisor=32),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='DefaultFormatBundle'),
    dict(
        type='Collect',
        keys=['img'])
]

# dataset settings
dataset_type = 'COCODataset'
data_root = 'data/coco/'
classes = ('a', 'b', 'c', 'd', 'b&d')
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type='CocoDataset',
        ann_file=
        '/data/xxx/annotations/train_data.json',
        img_prefix=
        '/data/xxx/images/',
        pipeline=train_pipeline,
        classes=classes),
    val=dict(
        type='CocoDataset',
        ann_file=
        '/data/xxx/annotations/validation_data.json',
        img_prefix=
        '/data/xxx/images/',
        pipeline=test_pipeline,
        classes=classes),
    test=dict(
        type='CocoDataset',
        ann_file=
        '/data/xxx/annotations/test_data.json',
        img_prefix=
        '/data/xxx/images/',
        pipeline=test_pipeline,
        classes=classes),
)

# model settings
n_classes = len(classes)
model = dict(
    roi_head=dict(
        bbox_head=[
            dict(
                type='Shared2FCBBoxHead',
                num_classes=n_classes),
            dict(
                type='Shared2FCBBoxHead',
                num_classes=n_classes),
            dict(
                type='Shared2FCBBoxHead',
                num_classes=n_classes)],
    mask_head=dict(num_classes=n_classes)))

work_dir = '/workspace/yyy/logs/mask_rcnn_r50_fpn_albu_1x_coco'

```
1. I ran the following command?

```bash
python /mmdetection/tools/train.py /workspace/yyy/configs/mask_rcnn_r50_fpn_albu_1x_coco.py
```

2. I made the modifications explained above to the config file. I understand the modifications I did.
3. I used a custom dataset of type `COCODataset`

**Environment**

1. Output of `python mmdet/utils/collect_env.py`:
```bash
sys.platform: linux
Python: 3.7.7 (default, May  7 2020, 21:25:33) [GCC 7.3.0]
CUDA available: True
GPU 0: Tesla V100-SXM2-32GB
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GCC: gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
PyTorch: 1.6.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.1 Product Build 20200208 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.7.0
OpenCV: 4.5.3
MMCV: 1.3.8
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.1
MMDetection: 2.16.0+fa10539
```
2. I didn't have to install PyTorch because I use a Dockerfile which begins with
```Dockerfile
ARG PYTORCH="1.6.0"
ARG CUDA="10.1"
ARG CUDNN="7"

FROM pytorch/pytorch:${PYTORCH}-cuda${CUDA}-cudnn${CUDNN}-devel

```
3.  Other enviroment variables:
```bash
# echo $PATH
/opt/conda/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
# echo $LD_LIBRARY_PATH
/usr/local/nvidia/lib:/usr/local/nvidia/lib64
```

**Error traceback**

```python
Traceback (most recent call last):
  File "/mmdetection/tools/train.py", line 194, in <module>
    main()
  File "/mmdetection/tools/train.py", line 190, in main
    meta=meta)
  File "/mmdetection/mmdet/apis/train.py", line 174, in train_detector
    runner.run(data_loaders, cfg.workflow)
  File "/opt/conda/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 127, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 47, in train
    for i, data_batch in enumerate(self.data_loader):
  File "/opt/conda/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 363, in __next__
    data = self._next_data()
  File "/opt/conda/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 989, in _next_data
    return self._process_data(data)
  File "/opt/conda/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 1014, in _process_data
    data.reraise()
  File "/opt/conda/lib/python3.7/site-packages/torch/_utils.py", line 395, in reraise
    raise self.exc_type(msg)
ValueError: Caught ValueError in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/torch/utils/data/_utils/worker.py", line 185, in _worker_loop
    data = fetcher.fetch(index)
  File "/opt/conda/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/opt/conda/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/mmdetection/mmdet/datasets/custom.py", line 195, in __getitem__
    data = self.prepare_train_img(idx)
  File "/mmdetection/mmdet/datasets/custom.py", line 218, in prepare_train_img
    return self.pipeline(results)
  File "/mmdetection/mmdet/datasets/pipelines/compose.py", line 41, in __call__
    data = t(data)
  File "/mmdetection/mmdet/datasets/pipelines/transforms.py", line 1462, in __call__
    results = self.aug(**results)
  File "/opt/conda/lib/python3.7/site-packages/albumentations/core/composition.py", line 188, in __call__
    p.preprocess(data)
  File "/opt/conda/lib/python3.7/site-packages/albumentations/core/utils.py", line 84, in preprocess
    data[data_name] = self.check_and_convert(data[data_name], rows, cols, direction="to")
  File "/opt/conda/lib/python3.7/site-packages/albumentations/core/utils.py", line 92, in check_and_convert
    return self.convert_to_albumentations(data, rows, cols)
  File "/opt/conda/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 51, in convert_to_albumentations
    return convert_bboxes_to_albumentations(data, self.params.format, rows, cols, check_validity=True)
  File "/opt/conda/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 306, in convert_bboxes_to_albumentations
    return [convert_bbox_to_albumentations(bbox, source_format, rows, cols, check_validity) for bbox in bboxes]
  File "/opt/conda/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 306, in <listcomp>
    return [convert_bbox_to_albumentations(bbox, source_format, rows, cols, check_validity) for bbox in bboxes]
  File "/opt/conda/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 254, in convert_bbox_to_albumentations
    check_bbox(bbox)
  File "/opt/conda/lib/python3.7/site-packages/albumentations/augmentations/bbox_utils.py", line 333, in check_bbox
    "to be in the range [0.0, 1.0], got {value}.".format(bbox=bbox, name=name, value=value)
ValueError: Expected x_max for bbox (0.8815625508626302, 0.4241667005750868, 1.7990626017252604, 0.8737500508626302, 0) to be in the range [0.0, 1.0], got 1.7990626017252604.
```

**Comment by hhaAndroid** [comment]
@AndreaPi The format parameter should be `pascal_voc `instead of `coco`. If the assert error still occurs, it means that there is an out-of-bounds situation in your label.

**Comment by AndreaPi** [comment]
@hhaAndroid thanks, that fixed that error, but **why** did ti work? My dataset is in `COCODataset` format, not PASCAL_VOC. 

However, now I have an `AttributeError`:

```
Traceback (most recent call last):
  File "/mmdetection/tools/train.py", line 194, in <module>
    main()
  File "/mmdetection/tools/train.py", line 190, in main
    meta=meta)
  File "/mmdetection/mmdet/apis/train.py", line 174, in train_detector
    runner.run(data_loaders, cfg.workflow)
  File "/opt/conda/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 127, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 50, in train
    self.run_iter(data_batch, train_mode=True, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 30, in run_iter
    **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/mmcv/parallel/data_parallel.py", line 67, in train_step
    return self.module.train_step(*inputs[0], **kwargs[0])
  File "/mmdetection/mmdet/models/detectors/base.py", line 238, in train_step
    losses = self(**data)
  File "/opt/conda/lib/python3.7/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/mmcv/runner/fp16_utils.py", line 97, in new_func
    return old_func(*args, **kwargs)
  File "/mmdetection/mmdet/models/detectors/base.py", line 172, in forward
    return self.forward_train(img, img_metas, **kwargs)
  File "/mmdetection/mmdet/models/detectors/two_stage.py", line 149, in forward_train
    **kwargs)
  File "/mmdetection/mmdet/models/roi_heads/standard_roi_head.py", line 105, in forward_train
    img_metas)
  File "/mmdetection/mmdet/models/roi_heads/standard_roi_head.py", line 134, in _bbox_forward_train
    bbox_results = self._bbox_forward(x, rois)
  File "/mmdetection/mmdet/models/roi_heads/standard_roi_head.py", line 124, in _bbox_forward
    cls_score, bbox_pred = self.bbox_head(bbox_feats)
  File "/opt/conda/lib/python3.7/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/torch/nn/modules/container.py", line 117, in forward
    input = module(input)
  File "/opt/conda/lib/python3.7/site-packages/torch/nn/modules/module.py", line 722, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/mmdetection/mmdet/models/roi_heads/bbox_heads/convfc_bbox_head.py", line 162, in forward
    x = x.flatten(1)
AttributeError: 'tuple' object has no attribute 'flatten' 
```

Since the error is different now, should I close this issue and open a new one?

**Comment by AndreaPi** [comment]
Ok, the `AttributeError` was due to the fact that I made a mistake when overwriting all the `num_classes` from default value (e.g. 80 in COCO) to my number of classes. The right way to do it is illustrated here:

  https://github.com/open-mmlab/mmdetection/issues/6141#issuecomment-924164079

I **still** can't get the Albumentation augmentations to work, but now the error I get is very different, so I'll close this issue and open a new one.
